version: '3'
services:
  dataact-broker-db:
    container_name: dataact-broker-db
    image: postgres:10.6-alpine
    restart: on-failure:3 # 3 max attempt, and then it will stop restarting
    environment:
      POSTGRES_USER: admin
      POSTGRES_PASSWORD: root
      POSTGRES_DB: data_broker
    ports:
      - "5435:5432"

  dataact-broker-init-db:
    container_name: dataact-broker-init-db
    image: dataact-broker-backend
    build:
      context: ./
    depends_on:
      - dataact-broker-db
    command: /bin/sh -x -c "sleep 9s; cd ./dataactcore; alembic upgrade head"
    volumes:
      - ./:/data-act/backend

  dataact-broker-backend:
    container_name: dataact-broker-backend
    image: dataact-broker-backend
    build:
      context: ./
    depends_on:
      - dataact-broker-db
    restart: on-failure:3 # 3 max attempt, and then it will stop restarting
    command: /bin/sh -c "sleep 9s; python dataactbroker/app.py"
    ports:
      - "9999:9999"
    volumes:
      - ./:/data-act/backend

  dataact-broker-validator:
    container_name: dataact-broker-validator
    image: dataact-broker-backend
    restart: on-failure:5 # 5 max attempt, and then it will stop restarting
    command: /bin/sh -c "sleep 17s; python dataactvalidator/app.py"
    ports:
      - "8889:8889"
    volumes:
      - ./:/data-act/backend
